crates/forge_analyzer/src/definitions.rs

```diff
@@ -2678,19 +3116,34 @@ impl Visit for ExportCollector<'_> {
             }
             DefaultDecl::TsInterfaceDecl(_) => {}
         }
+        n.decl.visit_children_with(self);
     }
 
     fn visit_export_named_specifier(&mut self, n: &ExportNamedSpecifier) {
-        let orig_id = n.orig.as_id();
-        let orig = self.add_export(DefRes::default(), orig_id);
-        if let Some(id) = &n.exported {
-            let exported_id = id.as_id();
-            self.add_export(DefRes::ExportAlias(orig), exported_id);
+        if let ModuleExportName::Ident(ident) = &n.orig {
+            let export_defid = self.add_export(DefRes::Undefined, ident.to_id());
+            self.res_table
+                .exported_names
+                .insert((ident.sym.clone(), self.curr_mod), export_defid);
+        } else {
+            let orig_id = n.orig.as_id();
+            let orig = self.add_export(DefRes::default(), orig_id);
+            if let Some(id) = &n.exported {
+                let exported_id = id.as_id();
+                self.add_export(DefRes::ExportAlias(orig), exported_id);
+            }
+            n.visit_children_with(self)
         }
     }
 
-    fn visit_export_default_expr(&mut self, _: &ExportDefaultExpr) {
-        self.add_default(DefRes::Undefined, None);
+    fn visit_export_default_expr(&mut self, n: &ExportDefaultExpr) {
+        if let Expr::Ident(ident) = &*n.expr {
+            self.add_default(DefRes::Undefined, None);
+            // adding the default export, so we can resolve it during the lowering
+            self.res_table
+                .exported_names
+                .insert((ident.sym.clone(), self.curr_mod), self.default.unwrap());
```

Comment:
Could you change the `ExportCollector::set_default` method to return the `DefId` that was set to `self.default` at the end of that function?; That way you can do `let default_id = self.add_default(DefRes::Undefined, None);` and replace `self.default.unwrap()` with `default_id`.

Reply:
There is no set_default function.

Reply:
Tldr on my efforts: For this one, I couldn't find `set_default` under ExportCollector. Nor did a `default_id` variable exist. 

So I tried to return defid in add_default. But that also messes up some self.add_default(DefRes::Undefined, None); implementations elsewhere because it's expected that nothing gets returned. 

I did try creating self.set_default, but it seemed a little repetitive to call it if add_default is already setting self.default. and calling set_default to return a defId add_default is pretty much the same result as the approach described above. 

Otherwise, the easiest way to not use .unwrap() might be to use a let statement grab the value in a Some() object instead of returning it. 

Do you want me to break a lil and adapt like other places where self.add_default is called?

Reply:
Ah, I meant change `add_default` to return a DefId. Yeah, that should be fine, since looking at the breakage, it's just match arms that are now returning a `DefId` instead of `()`. To fix it, you just need to wrap the arm in braces and add a semicolon.